src/main/java/org/openlmis/buq/domain/buq/BottomUpQuantification.java

```diff
@@ -0,0 +1,196 @@
+  @Getter
+  @Setter
+  private BottomUpQuantificationStatus status;
+
+  @OneToMany(
+      mappedBy = "bottomUpQuantification",
+      cascade = {CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH, CascadeType.REMOVE},
+      fetch = FetchType.LAZY,
+      orphanRemoval = true)
+  @Getter
+  @Setter
+  private List<BottomUpQuantificationLineItem> bottomUpQuantificationLineItems;
+
+  @OneToMany(
+      mappedBy = "bottomUpQuantification",
+      cascade = CascadeType.ALL)
+  @Getter
+  @Setter
+  private List<BottomUpQuantificationStatusChange> statusChanges = new ArrayList<>();
+
+  /**
+   * Constructor.
+   *
+   * @param facilityId         id of the Facility
+   * @param programId          id of the Program
+   * @param processingPeriodId id of the ProcessingPeriod
+   */
+  public BottomUpQuantification(UUID facilityId, UUID programId, UUID processingPeriodId) {
+    this.facilityId = facilityId;
+    this.programId = programId;
+    this.processingPeriodId = processingPeriodId;
+  }
+
+  /**
+   * Creates new instance based on data from the importer.
+   */
+  public static BottomUpQuantification newInstance(Importer importer) {
+    BottomUpQuantification buq = new BottomUpQuantification();
+
+    buq.setCreatedDate(importer.getCreatedDate());
+    buq.setModifiedDate(importer.getModifiedDate());
+    buq.setFacilityId(importer.getFacilityId());
+    buq.setProgramId(importer.getProgramId());
+    buq.setProcessingPeriodId(importer.getProcessingPeriodId());
+    buq.setTargetYear(importer.getTargetYear());
+    buq.setStatus(importer.getStatus());
+
+    return buq;
+  }
+
+  /**
+   * Copy values of attributes into new or updated BottomUpQuantification.
+   *
+   * @param original object to update from.
+   * @param lineItems list of bottom-up quantification line items.
+   */
+  public void updateFrom(BottomUpQuantification original,
+      List<BottomUpQuantificationLineItem> lineItems) {
+    setModifiedDate(ZonedDateTime.now());
+    setCreatedDate(original.getCreatedDate());
+
+    facilityId = original.getFacilityId();
+    programId = original.getProgramId();
+    processingPeriodId = original.getProcessingPeriodId();
+    targetYear = original.getTargetYear();
+    status = original.getStatus();
+
+    if (lineItems != null) {
+      bottomUpQuantificationLineItems.clear();
+      bottomUpQuantificationLineItems.addAll(lineItems);
+    }
+  }
+
+  /**
+   * Export this object to the specified exporter (DTO).
+   *
+   * @param exporter exporter to export to
+   */
+  public void export(Exporter exporter) {
+    exporter.setId(getId());
+    exporter.setCreatedDate(getCreatedDate());
+    exporter.setModifiedDate(getModifiedDate());
+    exporter.setFacilityId(facilityId);
+    exporter.setProgramId(programId);
+    exporter.setProcessingPeriodId(processingPeriodId);
+    exporter.setTargetYear(targetYear);
+    exporter.setStatus(status);
+  }
+
+  @PrePersist
+  private void initModifiedDate() {
```

Comment:
I think this is a reasonable default behavior of the `BaseTimestampedEntity` and could be moved there.

Reply:
I moved this method's logic to prePersist() method in BaseTimestampedEntity class.